chef/cookbooks/network/recipes/default.rb

```diff
@@ -151,6 +151,7 @@ def kill_nic(nic)
 
   Chef::Log.info("Interface #{nic.name} is no longer being used, deconfiguring it.")
   nic.destroy
+  ::Kernel.system("wicked ifdown #{nic.name}")
```

Comment:
isn't the order the wrong way around? why not first tell wicked that we're going to kill it and then call nic.destroy, e.g. swap the two lines above?

Reply:
Ok, I see the comment above now.